tfe_hardware/qcodes_instrument_drivers/Sinko/AD5760.py

```diff
@@ -0,0 +1,184 @@
+        manager = ClientManager.Create(-1)
+        self.client = manager.CreateRequestClient(f"localhost:{host_name}")
+        self.client.ContextPath = path
+        self.client.WriteRegister("2", "786")
+        
+    def _get_set_volt(self,
+                        output_level: Optional[float] = None
+                        ) -> Optional[float]:
+        """
+        Get or set the voltage output.
+
+        Args:
+            output_level: If missing, we assume that we are getting the
+                voltage output. Else we are setting it
+        """
+        if output_level is not None:
+            self.write(output_level)
+            return None
+        return self.ask()
+    
+    def _get_set_current(self,
+                      output_level: Optional[float] = None
+                      ) -> Optional[float]:
+        
+        """
+        Get or set the current output.
+
+        Args:
+            output_level: If missing, we assume that we are getting the
+                current output. Else we are setting it
+        """
+
+        if output_level is not None:
+            self.write(output_level*100)
+            return None
+        return self.ask()/100
+    
+    
+    def ramp_current(self, ramp_to: float, step: float, delay: float) -> None:
+        """
+        Ramp the current from the current level to the specified output.
+
+        Args:
+            ramp_to: The ramp target in Amps
+            step: The ramp steps in Amps
+            delay: The time between finishing one step and
+                starting another in seconds.
+        """
+        self.ramp_trial(ramp_to*100, step*100, delay)
+
+        
+    def ramp_voltage(self, ramp_to: float, step: float, delay: float) -> None:
+        """
+        Ramp the voltage from the current level to the specified output.
+
+        Args:
+            ramp_to: The ramp target in Volt
+            step: The ramp steps in Volt
+            delay: The time between finishing one step and
+                starting another in seconds.
+        """
+
+        self.ramp_trial(ramp_to, step, delay)
+
+
+    def ramp_trial(self, ramp_to: float, step: float, delay: float):
+        
+        v_in = self.voltage()
+        v_f = v_in
+        if ramp_to > v_in:
+            num_steps = (ramp_to - v_in)/step +1
+            for i in range(int(num_steps+1)):
+                v_f += step
+                self.voltage(v_f)
+                time.sleep(delay)
+        elif ramp_to < v_in:
+            num_steps = - (ramp_to - v_in)/step 
+            for i in range(int(num_steps+1)):
+                v_f -= step
+                self.voltage(v_f)
+                time.sleep(delay) 
+
+
+    def _get_volt(self):
+        return self.ask()
+    
+    def _get_current(self):
+        return self.ask()/100
+
+    def _set_output(self, output_level) -> None:
+        self.write()
+        
+    def ask(self):
+        '''
+        More features to be added
```

Comment:
here also explain what those numbers are doing, a link to some external documentation is good enough for this. Some clue for the next person to use to have some idea of what is happening here